apps/smbmount.sh

```diff
@@ -0,0 +1,296 @@
+if [ "$(stat -c %a /etc/fstab)" != "600" ]
+then
+    chmod 600 /etc/fstab
+fi
+# Variables
+SMBSHARES="/mnt/smbshares"
+SMBSHARES_SED="\/mnt\/smbshares"
+
+# Functions
+add_mount() {
+# Check if mounting slots are available
+if grep -q "$SMBSHARES/1" /etc/fstab && grep -q "$SMBSHARES/2" /etc/fstab && grep -q "$SMBSHARES/3" /etc/fstab
+then
+    msg_box "All three slots are occupied. No mounting slots available. Please delete one of the SMB-mounts."
+    return
+fi
+# Enter SMB-server and Share-name
+while true
+do
+    SERVER_SHARE_NAME=$(whiptail --inputbox "Please Enter the server and Share-name like this:\n//Server/Share" "$WT_HEIGHT" "$WT_WIDTH" 3>&1 1>&2 2>&3)
+    if [[ "no" == $(ask_yes_or_no "Is this correct? $SERVER_SHARE_NAME") ]]
+    then
+        msg_box "It seems like your weren't satisfied by the PATH you entered. Please try again."
+    else
+        SERVER_SHARE_NAME=${SERVER_SHARE_NAME// /\\040}
+        break
+    fi
+done
+# Enter the SMB-user
+while true
+do
+    SMB_USER=$(whiptail --inputbox "Please enter the username of the SMB-user" "$WT_HEIGHT" "$WT_WIDTH" 3>&1 1>&2 2>&3)
+    if [[ "no" == $(ask_yes_or_no "Is this correct? $SMB_USER") ]]
+    then
+        msg_box "It seems like your weren't satisfied by the SMB-user you entered. Please try again."
+    else
+        break
+    fi
+done
+# Enter the password of the SMB-user
+while true
+do
+    SMB_PASSWORD=$(whiptail --inputbox "Please enter the password of the SMB-user $SMB_USER" "$WT_HEIGHT" "$WT_WIDTH" 3>&1 1>&2 2>&3)
+    if [[ "no" == $(ask_yes_or_no "Is this correct? $SMB_PASSWORD") ]]
+    then
+        msg_box "It seems like your weren't satisfied by the password for the SMB-user you entered. Please try again."
+    else
+        break
+    fi
+done
+# Get the uid and gid of www-data
+UsID=$(id www-data | awk '{print $1}')
+UsID=${UsID//[!0-9]/}
+GrID=$(id www-data | awk '{print $2}')
+GrID=${GrID//[!0-9]/}
+# Write everything to /etc/fstab, mount and connect external storage
+count=1
+while  [ $count -le 3 ]
+do
+    # Check which mounting slot is available
+    if ! grep -q "$SMBSHARES/$count" /etc/fstab
+    then 
+        # Write to /etc/fstab and mount
+        echo "$SERVER_SHARE_NAME $SMBSHARES/$count cifs username=$SMB_USER,password=$SMB_PASSWORD,vers=3,uid=$UsID,gid=$GrID,file_mode=0770,dir_mode=0770,nounix,noserverino 0 0" >> /etc/fstab
+        mkdir -p "$SMBSHARES/$count"
+        mount "$SMBSHARES/$count"
+        # Check if mounting was successful
+        if [[ ! $(findmnt -M "$SMBSHARES/$count") ]]
+        then
+            # If not remove this line from fstab
+            msg_box "It seems like the mount wasn't successful. It will get deleted now. Please try again.\nAs a hint: you might fix the connection problem by enabling SMB3 on your SMB-server."
+            sed -i "/$SMBSHARES_SED\/$count/d" /etc/fstab
+            break
+        else
+            # Install and enable files_external
+            install_and_enable_app files_external
```

Comment:
actually the only thing, I find a bit odd is, that this function enables the external storage app each time or at least it says that to the user...
Maybe a check would be better, if the app is already enabled and only enable it if it is disabled.

Reply:
and btw: files_external doesn't has to get installed first (since it is shipped together with Nextcloud). so that check is unnecessary here.

Reply:
Please read the function.

There's no need for a check as we want to install it if its not installed, and enable it if its not enabled. That's what the function does. It will always say that it enables it even if its already enabled since that's how the `occ` command works in Nextcloud.

Adding a check is just cosmetic in my opinion as the only change would be that we don't show the "enables text" to the user. Also prone to more failure, because what if the `occ` command displays the info differently in coming versions? Then the check would fail anyway.

Reply:
OK